.github/workflows/ci.yml

```
@@ -0,0 +1,48 @@
+        run: cargo test --all --all-targets --locked
+
+  clippy:
+    name: Clippy lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@1.87.0
+        with:
+          components: clippy
+      - name: Run clippy
+        run: cargo clippy --all --all-targets -- -D warnings 
```

YAMLlint (1.37.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)